webdriver/webdriver.go

```diff
@@ -13,28 +15,71 @@ var (
 	startFrameBuffer = flag.Bool("frame_buffer", frameBufferDefault(), "Whether to use a frame buffer")
 	seleniumPath     = flag.String("selenium_path", "", "Path to the selenium standalone binary.")
 	seleniumHost     = flag.String("selenium_host", "localhost", "Host to run selenium on")
-	seleniumPort     = flag.Int("selenium_port", 8888, "Port to run selenium on")
+	seleniumPort     = flag.Int("selenium_port", 0, "Port to run selenium on")
 )
 
 func frameBufferDefault() bool {
 	return runtime.GOOS != "darwin"
 }
 
+func pickUnusedPort() int {
+	addr, err := net.ResolveTCPAddr("tcp", "127.0.0.1:0")
+	if err != nil {
+		panic(err)
+	}
+	l, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		panic(err)
+	}
+	port := l.Addr().(*net.TCPAddr).Port
+	if err := l.Close(); err != nil {
+		panic(err)
+	}
+	return port
+}
+
+// GetWebDriver starts a WebDriver service (server) and creates a remote
+// (client).
+// Make sure to close both the service and the WebDriver instances, e.g.
+//
+// server, driver, err := GetWebDriver()
+// if err != nil {
+//   panic(err)
+// }
+// defer server.Stop()
+// defer driver.Quit()
 func GetWebDriver() (*selenium.Service, selenium.WebDriver, error) {
+	if *seleniumPath == "" {
+		panic("-selenium_path not specified")
+	}
+	if *seleniumPort == 0 {
+		port := pickUnusedPort()
+		seleniumPort = &port
+	}
+
+	var options []selenium.ServiceOption
+	if *startFrameBuffer {
+		options = append(options, selenium.StartFrameBuffer())
+	}
+	options = append(options, selenium.Output(os.Stderr))
+
 	switch *browser {
 	case "firefox":
-		return FirefoxWebDriver()
+		return FirefoxWebDriver(options)
 	case "chrome":
-		return ChromeWebDriver()
+		return ChromeWebDriver(options)
 	}
-	panic("Invalid --browser value specified")
+	panic("Invalid -browser value specified")
 }
 
 // FindShadowElements finds the shadow DOM children via the given query
-// selectors, recursively.
-// e.g. FindShadowElements(wd, foo, "bar", "baz") would be similar to
-// A "foo bar baz" CSS selector, except it crosses the shadow boundaries for
-// each separate selector.
+// selectors, recursively. The function takes a variable number of selectors;
+// the selectors are combined together similar to CSS descendant combinators.
+// However, all but the the last selector are expected to match to hosts of
+// shadow DOM, and the shadow DOM boundaries will be crossed.
+// e.g. FindShadowElements(wd, node, "bar", "baz blah"). All matches of "bar"
```

Comment:
nit: Add nothing-but-`//` line before e.g. to break large comment into logical chunks.